### ISSUE TITLE
I got AP_{Class_name} always equal 0.0, Is there anyone got the result like me and Why is that?

### ISSUE THREAD

**Issue by JK-Pair** [google/automl#753]

**Comment by Steve-2040** [comment]
@JK-Pair There seems to be an open issue #676 with custom dataset. It may help to add 'use_keras_model: False' to the yaml file.

**Comment by JK-Pair** [comment]
> @JK-Pair There seems to be an open issue #676 with custom dataset. It may help to add 'use_keras_model: False' to the yaml file and see if that makes a difference.

I already changed 'use_keras_model' = False, I got the same results.

**Comment by Steve-2040** [comment]
I do my own basic evaluation so, can’t help with other methods.
I only train then use the savedmodel and this basic script to compare the inference outputs to the expected outputs.
test images are labeled img0.jpg to img99.jpg in directory test_inf and saved model is in savedmodeldir
```python
import tensorflow as tf
from PIL import Image, ImageOps
import numpy as np
import os

# os.environ['CUDA_VISIBLE_DEVICES'] = "-1" # uncomment to run on CPU

gpus = tf.config.experimental.list_physical_devices('GPU')
if gpus:
    tf.config.experimental.set_memory_growth(gpus[0], True)
print(len(gpus), "Physical GPUs")

imported = tf.saved_model.load('savedmodeldir/')
f = imported.signatures["serving_default"]

def get_image_array(im_id):
    im = Image.open(f"test_inf/img{im_id}.jpg")
    im_arr = np.frombuffer(im.tobytes(), dtype=np.uint8)
    im_arr = im_arr.reshape((1, im.size[1], im.size[0], 3))
    return im_arr

for i in range(0, 100):
    im_arr = get_image_array(i)
    detections = f(tf.constant(im_arr))['detections:0'].numpy()
    detections = detections[0]
    y1 = int(detections[0][1])
    x1 = int(detections[0][2])
    y2 = int(detections[0][3])
    x2 = int(detections[0][4])
    pr = int(detections[0][5]*100.0) # probability
    cl = int(detections[0][6]) # class_id
            
    print(f"img{i}.jpg class_id:{cl} probability%:{pr} y1:{y1} x1:{x1} y2:{y2} x2:{x2}")
```

**Comment by mingxingtan** [comment]
AP_{class name} has some bugs. I will try to fix the bug later.
How about APs, APm, APl, AP50?

**Comment by JK-Pair** [comment]
> I do my own basic evaluation so, can’t help with other methods.
> I only train then use the savedmodel and this basic script to compare the inference outputs to the expected outputs.
> test images are labeled img0.jpg to img99.jpg in directory test_inf and saved model is in savedmodeldir
> 
> ```python
> import tensorflow as tf
> from PIL import Image, ImageOps
> import numpy as np
> import os
> 
> # os.environ['CUDA_VISIBLE_DEVICES'] = "-1" # uncomment to run on CPU
> 
> gpus = tf.config.experimental.list_physical_devices('GPU')
> if gpus:
>     tf.config.experimental.set_memory_growth(gpus[0], True)
> print(len(gpus), "Physical GPUs")
> 
> imported = tf.saved_model.load('savedmodeldir/')
> f = imported.signatures["serving_default"]
> 
> def get_image_array(im_id):
>     im = Image.open(f"test_inf/img{im_id}.jpg")
>     im_arr = np.frombuffer(im.tobytes(), dtype=np.uint8)
>     im_arr = im_arr.reshape((1, im.size[1], im.size[0], 3))
>     return im_arr
> 
> for i in range(0, 100):
>     im_arr = get_image_array(i)
>     detections = f(tf.constant(im_arr))['detections:0'].numpy()
>     detections = detections[0]
>     y1 = int(detections[0][1])
>     x1 = int(detections[0][2])
>     y2 = int(detections[0][3])
>     x2 = int(detections[0][4])
>     pr = int(detections[0][5]*100.0) # probability
>     cl = int(detections[0][6]) # class_id
>             
>     print(f"img{i}.jpg class_id:{cl} probability%:{pr} y1:{y1} x1:{x1} y2:{y2} x2:{x2}")
> ```

**Comment by JK-Pair** [comment]
> AP_{class name} has some bugs. I will try to fix the bug later.
> How about APs, APm, APl, AP50?

Other AP is works! except AP_{classname}

**Comment by JK-Pair** [comment]
> I do my own basic evaluation so, can’t help with other methods.
> I only train then use the savedmodel and this basic script to compare the inference outputs to the expected outputs.
> test images are labeled img0.jpg to img99.jpg in directory test_inf and saved model is in savedmodeldir
> 
> ```python
> import tensorflow as tf
> from PIL import Image, ImageOps
> import numpy as np
> import os
> 
> # os.environ['CUDA_VISIBLE_DEVICES'] = "-1" # uncomment to run on CPU
> 
> gpus = tf.config.experimental.list_physical_devices('GPU')
> if gpus:
>     tf.config.experimental.set_memory_growth(gpus[0], True)
> print(len(gpus), "Physical GPUs")
> 
> imported = tf.saved_model.load('savedmodeldir/')
> f = imported.signatures["serving_default"]
> 
> def get_image_array(im_id):
>     im = Image.open(f"test_inf/img{im_id}.jpg")
>     im_arr = np.frombuffer(im.tobytes(), dtype=np.uint8)
>     im_arr = im_arr.reshape((1, im.size[1], im.size[0], 3))
>     return im_arr
> 
> for i in range(0, 100):
>     im_arr = get_image_array(i)
>     detections = f(tf.constant(im_arr))['detections:0'].numpy()
>     detections = detections[0]
>     y1 = int(detections[0][1])
>     x1 = int(detections[0][2])
>     y2 = int(detections[0][3])
>     x2 = int(detections[0][4])
>     pr = int(detections[0][5]*100.0) # probability
>     cl = int(detections[0][6]) # class_id
>             
>     print(f"img{i}.jpg class_id:{cl} probability%:{pr} y1:{y1} x1:{x1} y2:{y2} x2:{x2}")
> ```

Thank you for your help, I implemented this code with my task, It works for me.

**Comment by chinaandylee** [comment]
I got a similar problem, some of AP_{Class_name} always equal -1, APs equal -1,  ARs  equal  -1.0,but others works.

Accumulating evaluation results...
DONE (t=0.68s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.900
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.984
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.984
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.865
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.941
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.773
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.921
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.921
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.892
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.955

INFO:tensorflow:Saving dict for global step 120000: AP = 0.9002168, AP50 = 0.98377615, AP75 = 0.98377615, AP_/bs-jpedA = 0.8540089, .... AP_/hhl-ytxmlB = -1.0, AP_/lq-yxhB = 0.93774754, AP_/lz-xzzpA = -1.0, AP_/lz-xzzpB = 0.7520792, ..., APl = 0.94077784, APm = 0.8646704, APs = -1.0, ARl = 0.9548545, ARm = 0.8916112, ARmax1 = 0.7728434, ARmax10 = 0.92097265, ARmax100 = 0.92097265, ARs = -1.0, box_loss = 0.00033786093, cls_loss = 0.158024, global_step = 120000, loss = 0.7840427
